lib/AST/UnqualifiedLookup.cpp

```diff
@@ -58,19 +58,34 @@ namespace {
     // lookup.
     if (loc.isInvalid() || decl->getBraces().isInvalid())
       return true;
-    
+
+    SourceManager &SM = decl->getASTContext().SourceMgr;
+
+    // If a code completion happens inside a function body, some lookups may
+    // happen from the 'loc' that is in a different buffer from the 'decl'.
+    // In such cases, look for members because we know 'loc' is inside a
+    // function body in the 'decl'.
+    if (SM.getCodeCompletionBufferID() != 0U) {
```

Comment:
I'd suggest a convenience API `hasCodeCompletionBuffer()` so that the caller doesn't have to remember what return value to check for.

Reply:
That's a great suggestion! I second it.